JustLog/Classes/LogstashDestination.swift

```diff
@@ -60,31 +78,45 @@ public class LogstashDestination: BaseDestination  {
     }
 
     public func forceSend(_ completionHandler: @escaping (_ error: Error?) -> Void  = {_ in }) {
-        
-        if self.logsToShip.count == 0 || self.socketManager.isConnected() {
-            completionHandler(nil)
-            return
-        }
 
         self.completionHandler = completionHandler
-        
-        logDispatchQueue.addOperation { [weak self] in
-            self?.socketManager.send()
-        }
+
+        writeLogs()
     }
     
     func writeLogs() {
         
         self.logDispatchQueue.addOperation{ [weak self] in
             
-            guard let `self` = self else { return }
+            guard let self = self else { return }
             
-            for log in self.logsToShip.sorted(by: { $0.0 < $1.0 }) {
-                let logData = self.dataToShip(log.1)
-                self.socketManager.write(logData, withTimeout: self.socketManager.timeout, tag: log.0)
+            let logsInQueue = self.logsInSocketQueue
```

Comment:
if we need the `logsInQueue` only to check if an operation is pending I would suggest to change the name in something like `pendingLogs` and, more important, to change it from an array to a set. This is to reduce the complexity in the lookup in the `logsToShip.filter` line from quadratic to linear. I think that in case there are many logs in the queue this change can save us a lot of CPU utilization.